### ISSUE TITLE
Internal error in src/full/Agda/TypeChecking/Coverage/Match.hs:312

### ISSUE THREAD

**Issue by xekoukou** [agda/agda#2874]
Trying to case split on ```eq``` [here](https://github.com/xekoukou/sparrow/blob/7211b25a7744d0b5af6d6b4ad6bacfea08a6501e/agda/SetLLProp.agda#L161) results in an internal error in the above location.

Replacing ```eq``` with ```()``` typechecks. I assume that error is in the case split itself.

This is the aforementioned function : 
```agda
  compl-repl⇒id-abs : ∀ {i u} {l : LinLogic i {u}} {r : LinLogic i}
                      {il} {rll : LinLogic i} {vs : SetLL rll} d
                      (ind : IndexLL rll (pickLL d l r)) →
                    (dw : MSetLL (pickLL d l r)) → (deq : dw ≡ del ↓ ind)
                    (w : MSetLL (l < il > r)) (eq : w ≡ (pickLLₛ-sbcm {il = il} d dw (¬∅ (pickLLₛ (~ict d) ↓ ↓)))) (w₁ : SetLL (l < il > r)) →
                    w ≡ ¬∅ w₁ →
                    mapₛ (λ s → sic d (s-extend ind s)) (complLₛ vs) ≡ complLₛ (replacePartOf-abs {b = vs} (ic d ind) w)
  compl-repl⇒id-abs ic← ind ∅ deq .(¬∅ ↓) () ↓ refl
  compl-repl⇒id-abs ic→ ind ∅ deq .(¬∅ ↓) () ↓ refl
  compl-repl⇒id-abs {vs = vs} ic← ↓ ∅ deq .(¬∅ (sic ic→ ↓)) refl (sic .ic→ .↓) refl = compl-repl⇒id-abs1 (complLₛ vs)
  compl-repl⇒id-abs ic← (ic d ind) ∅ deq .(¬∅ (sic ds w1)) eq (sic ds w1) refl = ⊥-elim (∅-neq-¬∅ (trans deq (proj₂ (pickLLₛ-sbcm¬∅ d (inj₂ (_ , refl)))))) 
  compl-repl⇒id-abs {vs = vs} ic→ ↓ ∅ deq .(¬∅ (sic ic← ↓)) refl (sic .ic← .↓) refl = compl-repl⇒id-abs2 (complLₛ vs)
  compl-repl⇒id-abs {vs = vs} ic→ (ic d ind) ∅ deq .(¬∅ (sic ic← ↓)) refl (sic .ic← .↓) refl = ⊥-elim (∅-neq-¬∅ (trans deq (proj₂ (pickLLₛ-sbcm¬∅ d (inj₂ (_ , refl))))))
  compl-repl⇒id-abs ic← ind ∅ deq .(¬∅ (sbc w1 w2)) eq (sbc w1 w2) refl = {!!}
  compl-repl⇒id-abs ic→ ind ∅ deq .(¬∅ (sbc w1 w2)) eq (sbc w1 w2) refl = {!!}
  compl-repl⇒id-abs d ind (¬∅ x) deq .(¬∅ w1) eq w1 refl = {!!}
```

The context in the hole is this one : 
```agda
Goal: mapₛ (λ s → sic ic← (s-extend ind s)) (complLₛ .vs) ≡
      sbcm (complLₛ (`replacePartOf w1 to .vs at ind)) (complLₛ w2)
————————————————————————————————————————————————————————————
eq   : ¬∅ (sbc w1 w2) ≡ ¬∅ (sic ic→ ↓)
deq  : ∅ ≡ del ↓ ind
ind  : IndexLL .rll .l
.vs  : SetLL .rll
w2   : SetLL .r
w1   : SetLL .l
.rll : LinLogic .i
.r   : LinLogic .i
.il  : LLCT
.l   : LinLogic .i
.u   : Level
.i   : Size
```

My agda version is at this [commit](https://github.com/agda/agda/tree/e12fec58047ffae17084d8a9c06a98410794ddc6)

**Comment by andreasabel** [comment]
@jespercockx : Does PR #2866 fix this issue?

**Comment by jespercockx** [comment]
Yes! With my version `eq` is replaced by an absurd pattern.

**Comment by ice1000** [comment]
Maybe an off-topic question but I don't know where to ask: when will Agda 2.5.4 out? Just curious :sweat_smile:

**Comment by andreasabel** [comment]
I think we should aim for a pre-AIM release, where the next AIM is 28 Jan - 3 Feb in Budapest.

**Comment by andreasabel** [comment]
@jespercockx : So we can justify an early merge of your PR #2866 as fix for #2874. ;-)
